tests/test-infra/setup_azure.sh

```diff
@@ -77,5 +78,27 @@ if $USE_SERVICEBUS ; then
     --namespace=$DAPR_NAMESPACE \
     --from-literal=connectionString=${SERVICEBUS_CONNSTRING}
 else
-  echo "NOT configuring Service Bus as pubsub store"
+  echo "NOT configuring Service Bus as pubsub broker"
+fi
+
+if $USE_KEYVAULT ; then
+  echo "Configuring Azure Key Vault as crypto provider"
+
+  # Set environmental variables to Azure Key Vault crypto provider
+  export DAPR_TEST_CRYPTO=azurekeyvault
+  if [ -n "$GITHUB_ENV" ]; then
+    echo "DAPR_TEST_CRYPTO=${DAPR_TEST_CRYPTO}" >> $GITHUB_ENV
+  fi
+
+  AzureKeyVaultTenantId=$(echo $AZURE_CREDENTIALS | jq -r '.tenantId')
+  AzureKeyVaultClientId=$(echo $AZURE_CREDENTIALS | jq -r '.clientId')
+  AzureKeyVaultClientSecret=$(echo $AZURE_CREDENTIALS | jq -r '.clientSecret')
+
+  kubectl create secret generic azurekeyvault-secret \
+    --namespace=$DAPR_NAMESPACE \
+    --from-literal="tenant-id=${AzureKeyVaultTenantId}" \
+    --from-literal="client-id=${AzureKeyVaultClientId}" \
+    --from-literal="client-secret=${AzureKeyVaultClientSecret}"
```

Comment:
@ItalyPaleAle Why are the credentials being stored here?

Reply:
We need a service principal to authenticate with Azure Key Vault